parser/ast/ddl.go

```diff
@@ -2138,59 +2147,73 @@ const (
-		case ResourcePriority:
-			ctx.WriteKeyWord("PRIORITY ")
-			ctx.WritePlain("= ")
-			ctx.WriteKeyWord(model.PriorityValueToName(n.UintValue))
-		case ResourceUnitCPU:
-			ctx.WriteKeyWord("CPU ")
-			ctx.WritePlain("= ")
-			ctx.WriteString(n.StrValue)
-		case ResourceUnitIOReadBandwidth:
-			ctx.WriteKeyWord("IO_READ_BANDWIDTH ")
-			ctx.WritePlain("= ")
-			ctx.WriteString(n.StrValue)
-		case ResourceUnitIOWriteBandwidth:
-			ctx.WriteKeyWord("IO_WRITE_BANDWIDTH ")
-			ctx.WritePlain("= ")
-			ctx.WriteString(n.StrValue)
-		case ResourceBurstableOpiton:
-			ctx.WriteKeyWord("BURSTABLE ")
-			ctx.WritePlain("= ")
-			ctx.WritePlain(strings.ToUpper(fmt.Sprintf("%v", n.BoolValue)))
-		case ResourceGroupRunaway:
-			ctx.WritePlain("QUERY_LIMIT ")
-			ctx.WritePlain("= ")
-			if len(n.ResourceGroupRunawayOptionList) > 0 {
-				ctx.WritePlain("(")
-				for i, option := range n.ResourceGroupRunawayOptionList {
-					if i > 0 {
-						ctx.WritePlain(" ")
-					}
-					if err := option.Restore(ctx); err != nil {
-						return errors.Annotatef(err, "An error occurred while splicing CreateResourceGroupStmt Option: [%v]", i)
-					}
+	switch n.Tp {
+	case ResourceRURate:
+		ctx.WriteKeyWord("RU_PER_SEC ")
+		ctx.WritePlain("= ")
+		ctx.WritePlainf("%d", n.UintValue)
+	case ResourcePriority:
+		ctx.WriteKeyWord("PRIORITY ")
+		ctx.WritePlain("= ")
+		ctx.WriteKeyWord(model.PriorityValueToName(n.UintValue))
+	case ResourceUnitCPU:
+		ctx.WriteKeyWord("CPU ")
+		ctx.WritePlain("= ")
+		ctx.WriteString(n.StrValue)
+	case ResourceUnitIOReadBandwidth:
+		ctx.WriteKeyWord("IO_READ_BANDWIDTH ")
+		ctx.WritePlain("= ")
+		ctx.WriteString(n.StrValue)
+	case ResourceUnitIOWriteBandwidth:
+		ctx.WriteKeyWord("IO_WRITE_BANDWIDTH ")
+		ctx.WritePlain("= ")
+		ctx.WriteString(n.StrValue)
+	case ResourceBurstableOpiton:
+		ctx.WriteKeyWord("BURSTABLE ")
+		ctx.WritePlain("= ")
+		ctx.WritePlain(strings.ToUpper(fmt.Sprintf("%v", n.BoolValue)))
+	case ResourceGroupRunaway:
+		ctx.WritePlain("QUERY_LIMIT ")
+		ctx.WritePlain("= ")
+		if len(n.RunawayOptionList) > 0 {
+			ctx.WritePlain("(")
+			for i, option := range n.RunawayOptionList {
+				if i > 0 {
+					ctx.WritePlain(" ")
+				}
+				if err := option.Restore(ctx); err != nil {
+					return errors.Annotatef(err, "An error occurred while splicing CreateResourceGroupStmt Option: [%v]", i)
 				}
-				ctx.WritePlain(")")
-			} else {
-				ctx.WritePlain("NULL")
 			}
-		default:
-			return errors.Errorf("invalid ResourceGroupOption: %d", n.Tp)
+			ctx.WritePlain(")")
+		} else {
+			ctx.WritePlain("NULL")
 		}
-		return nil
+	case ResourceGroupBackground:
+		ctx.WritePlain("BACKGROUND ")
+		ctx.WritePlain("= ")
+		if len(n.BackgroundOptions) > 0 {
+			ctx.WritePlain("(")
+			for i, option := range n.BackgroundOptions {
+				if i > 0 {
+					ctx.WritePlain(", ")
+				}
+				if err := option.Restore(ctx); err != nil {
+					return errors.Annotatef(err, "An error occurred while splicing ResourceGroup Background Option")
```

Comment:
Do we need to unify error with `ResourceGroupRunaway` to expose `i`?
```suggestion
					return errors.Annotatef(err, "An error occurred while splicing ResourceGroup Background Option: [%v]", i")
```